lib/resource-loader.js

```diff
@@ -136,36 +144,45 @@ function loadFiles (resourceList, config) {
 
 function findFiles (config) {
   return new Promise((resolve, reject) => {
-    fs.readdir(`${config.projectLocation}/${config.nodeshiftDirectory}`, (err, files) => {
-      // Probably can ignore, or maybe have a warning or something if the directory doesn't exist
-      if (err && err.code === 'ENOENT') {
-        logger.warning(`No ${config.nodeshiftDirectory} directory`);
-        return resolve([]);
-      }
+    fs.readdir(
+      config.resourceProfile
+        ? `${config.projectLocation}/${config.nodeshiftDirectory}/${config.resourceProfile}`
+        : `${config.projectLocation}/${config.nodeshiftDirectory}`,
+      (err, files) => {
+        // Probably can ignore, or maybe have a warning or something if the directory doesn't exist
+        if (err && err.code === 'ENOENT') {
+          logger.warning(
+            config.resourceProfile
+              ? `No ${config.nodeshiftDirectory}/${config.resourceProfile} directory`
+              : `No ${config.nodeshiftDirectory} directory`
+          );
+          return resolve([]);
+        }
 
-      if (err && err.code !== 'ENOENT') {
-        return reject(err);
-      }
+        if (err && err.code !== 'ENOENT') {
+          return reject(err);
+        }
 
-      // Map the files into an object that has the ext and fileLocation
-      // Then filter since we only want the .yml or .yaml or .json files
-      const filtered = files.map((file) => {
-        // this will split the filename [0] and extension [1]
-        const filesSplit = file.split('.');
-
-        // this will split the filename into name [0] and type [1]
-        const filenameSplit = filesSplit[0].split('-');
-        return {
-          ext: filesSplit[1],
-          name: filenameSplit[0],
-          type: filenameSplit[1],
-          filename: file
-        };
-      }).filter((file) => {
-        return file.ext === 'yml' || file.ext === 'yaml' || file.ext === 'json';
+        // Map the files into an object that has the ext and fileLocation
+        // Then filter since we only want the .yml or .yaml or .json files
+        const filtered = files.map((file) => {
+          // this will split the filename [0] and extension [1]
+          const filesSplit = file.split('.');
+
+          // this will split the filename into name [0] and type [1]
+          const filenameSplit = filesSplit[0].split('-');
```

Comment:
what happens if there is a resourceProfile directory,  but the user doesn't specify it?  Will this call error or will it be ok,  and the `[1]` of in the code below will just be undefined and the filter will then ignore them

Reply:
I thought we wanted to look for a subdirectory only when the resourcePath flag is specified by the user otherwise, we are going to load whatever file we find in .nodeshift directory

Reply:
that is correct,  but there could be a sub directory(for a different profile) as well as files at the top level `.nodeshift` directoy,  and if they don't specify a resourceProfile, then just those files at the top level should be loaded.  and the directory should be ignored.

it is possible that nothing needs to change,  i forgot the behavior of accessing an array position that doesn't exist

Reply:
Maybe just added a test with that. "files at the top level with a subdirectory that has files but no resourceProfile flag used"